### PR TITLE
Add some S3 Examples

### PR DESCRIPTION
**Pull request by cooperlees**
- Show how to upload, check if an object exists + stream a download to chunk server without blocking indefinately.

P.s. I guessed `.rst` as best I could.

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -82,6 +82,143 @@ Use the batch writer to take care of dynamodb writing retries etc...
...
+
+.. code-block:: python3
+
+    async def upload(
+        suite: str,
+        release: str,
+        filename: str,
+        staging_path: Path,
+        bucket: str,
+        aws_secret_access_key: str,
```
Comment (terrycain) [inline docs/usage.rst]: As a simple example we can do without specifying keys

File: docs/usage.rst [hunk 2]
```diff
@@ -82,6 +82,143 @@ Use the batch writer to take care of dynamodb writing retries etc...
...
+        aws_access_key_id: str,
+    ) -> str:
+        blob_s3_key = f"{suite}/{release}/{filename}"
+        if not staging_path.exists():
+            LOG.error(
```
Comment (terrycain) [inline docs/usage.rst]: these docs are going to be displayed in the browser, we can get away with longer lines here so feel free to lengthen them if it aids readability

File: docs/usage.rst [hunk 3]
```diff
@@ -82,6 +82,143 @@ Use the batch writer to take care of dynamodb writing retries etc...
...
+        filename: str,
+        bucket: str,
+        aws_secret_access_key: str,
+        aws_access_key_id: str,
+    ) -> bool:
+        blob_s3_key = f"{suite}/{release}/{filename}"
+        async with aioboto3.client(
+            "s3",
+            aws_secret_access_key=aws_secret_access_key,
+            aws_access_key_id=aws_access_key_id,
```
Comment (terrycain) [inline docs/usage.rst]: as above

File: docs/usage.rst [hunk 4]
```diff
@@ -82,6 +82,143 @@ Use the batch writer to take care of dynamodb writing retries etc...
...
+        ) as s3:
+            object_list = await s3.list_objects_v2(
```
Comment (terrycain) [inline docs/usage.rst]: If your making a function to check if an object exists, use `client.head_object()`

File: docs/usage.rst [hunk 5]
```diff
@@ -82,6 +82,143 @@ Use the batch writer to take care of dynamodb writing retries etc...
...
+
+    from aiohttp import web
+    from multidict import MultiDict
+
+    async def serve_blob(
+        suite: str,
+        release: str,
+        filename: str,
+        bucket: str,
+        aws_secret_access_key: str,
```
Comment (terrycain) [inline docs/usage.rst]: same as above

File: docs/usage.rst [hunk 6]
```diff
@@ -82,6 +82,143 @@ Use the batch writer to take care of dynamodb writing retries etc...
...
+        aws_access_key_id: str,
+        request: web.Request,
+        chunk_size: int = 69
+    ) -> web.Response:
+        blob_s3_key = f"{suite}/{release}/{filename}"
+
+        if not await blob_exists(suite, release, filename):
+            # Not included but generate an error response for the HTTP Restful API
+            return self._gen_download_fnf_error(blob_s3_key)
```
Comment (terrycain) [inline docs/usage.rst]: This means nothing to the user, also there's no self in the parameter list.

File: docs/usage.rst (hunk 4 above)
Comment (cooperlees) [reply docs/usage.rst]: I don't follow - I'll remove this function since it seems I'm going about it a less optimal way.

**Comment by cooperlees** [comment]
Ok done. Can remove the logging but I feel the logging can help with understanding the code. Will remove if you prefer that.

**Comment by terrycain** [comment]
Nice thanks :smile:

**Pull request merged**